addons/mail/static/src/messaging/component/composer_text_input/composer_text_input.js

```diff
@@ -178,38 +294,116 @@ class ComposerTextInput extends Component {
      * @param {KeyboardEvent} ev
      */
     _onKeydownTextareaEnter(ev) {
-        if (!this.props.hasSendOnEnterEnabled) {
-            return;
+        if (this.composer.suggestedPartners.length > 0) {
+            ev.preventDefault();
+        } else {
+            if (!this.props.hasSendOnEnterEnabled) {
+                return;
+            }
+            if (ev.shiftKey) {
+                return;
+            }
+            if (this.env.messaging.device.isMobile) {
+                return;
+            }
+            this.trigger('o-keydown-enter');
+            ev.preventDefault();
         }
-        if (ev.shiftKey) {
-            return;
-        }
-        if (this.env.messaging.device.isMobile) {
-            return;
+    }
+
+    /**
+     * Key events management is performed in a Keyup to avoid intempestive RPC calls
+     *
+     * @private
+     * @param ev
+     */
+    _onKeyupTextarea(ev) {
+        switch (ev.key) {
+            // ESCAPE: close mention suggestions
+            case 'Escape':
+                this._onKeyupTextareaEscape(ev);
+                break;
+            // ENTER, HOME, END, UP, DOWN, PAGE UP, PAGE DOWN, TAB: check if navigation in mention suggestions
+            case 'Enter':
+                if (this.composer.suggestedPartners.length > 0) {
+                    if (this.composer.activePartner !== undefined) {
+                        this.composer.insertSuggestedPartner(this._mentionKeyword);
+                    }
+                }
+                break;
+            case 'ArrowUp':
+            case 'PageUp':
+                if (this.composer.suggestedPartners.length > 0) {
+                    this.composer.activatePreviousSuggestedPartner();
+                }
+                break;
+            case 'ArrowDown':
+            case 'PageDown':
+                if (this.composer.suggestedPartners.length > 0) {
+                    this.composer.activateNextSuggestedPartner();
+                }
+                break;
+            case 'Home':
+                if (this.composer.suggestedPartners.length > 0) {
+                    this.composer.activateFirstSuggestedPartner();
+                }
```

Comment:
Most of those events need a `preventDefault()`, in particular those that otherwise have effect on the cursor position: 
arrows, home, end, page up&down, ...
Probably tab too, because it would otherwise change the focus, ...

Reply:
There is a ```preventDefault()```, it's just called previously in the flow (at the keyDown).

Reply:
But page up&down, home and end were missing, I added them